### PR TITLE
Deploy integration tests - back ports

### PR DESCRIPTION
**Pull request by SimonRichardson**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

*Please replace with a description about why this change is needed, along with a description of what changed?*

## QA steps

```sh
cd tests && ./main.sh deploy
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/OH28fCR0/2116-make-27-branch-look-like-develop-in-terms-of-integration-tests

File: tests/main.sh [hunk 1]
```diff
@@ -25,13 +27,27 @@ import_subdir_files() {
...
 # If adding a test suite, then ensure to add it here to be picked up!
-TEST_NAMES="test_static_analysis \
-            test_branches \
-            test_cli \
-            test_controller \
-            test_cmr_bundles \
-            test_machine \
-            test_smoke"
+TEST_NAMES="static_analysis \
+            appdata \
```
Comment (hmlanigan) [inline tests/main.sh]: appdata and model do not exist in 2.7 as suites.

Comment (SimonRichardson) [reply tests/main.sh]: I need to back port more, to bring in better checking!

Comment (hmlanigan) [reply tests/main.sh]: Is there a bug, or card for this?  Or remove them now?

Comment (SimonRichardson) [reply tests/main.sh]: ah, yeah, good point

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**